source/reference/configure-shell-settings.txt

```diff
@@ -0,0 +1,166 @@
+.. _mongosh-shell-settings:
+
+=====================
+Configure ``mongosh``
+=====================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+
+The ``config`` API provides methods to examine and update the
+:mongosh:`mongosh </>` configuration. Updates made using ``config``
+persist between sessions.
+
+.. note::
+
+   The ``config`` API can be used within the ``mongosh`` command line
+   interface. It has no effect in the
+   :compass:`embedded Compass shell </embedded-shell>`.
+
+Syntax
+------
+
+Print the current ``mongosh`` configuration:
+
+.. code-block:: javascript
+
+   config 
+
+Return the current value for ``<property>``:
+
+.. code-block:: javascript
+
+   config.get( "<property>" )
+
+Change the current setting of ``<property>`` to ``<value>``:
+
+.. code-block:: javascript
+
+   config.set( "<property>", <value> )
+
+Supported ``property`` parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 15 10 50
+
+   * - Key
+     - Type
+     - Default
+     - Description
+
+   * - ``batchSize``
+     - integer
+     - 20
+     - The number of items displayed per cursor iteration
+
+   * - ``enableTelemetry``
+     - boolean
+     - false
+     - Enables sending annonymized tracking and diagnostic data to
+       MongoDB. 
+
+   * - ``historyLength``
+     - integer
+     - 1000
+     - The number of items to store in ``mongosh`` REPL's history file.
+
+   * - ``inspectDepth``
+     - integer or "infinity"
```

Comment:
Not sure about this one, I have two questions. 
1. It seems that almost any string will set infinity (other than something like "3" which sets integer 3). Is this expected? 
2. Depth of inspection works as expected, but the .pretty() modifier doesn't work consistently when inspectDepth is changed. Is this expected?

See   https://gist.github.com/davemungo/1b44d6f6316edc07eb3df9a858160420 for various examples.

Reply:
Regarding 1.: I don’t think it’s expected that you plug in strings where a numeric value is expected – we could probably provide a better error message in that case, though.

Regarding 2.: This is a bit related to https://jira.mongodb.org/browse/MONGOSH-684, but currently, `.pretty()` is a no-op in mongosh.

Reply:
@addaleax  I think I added this just after you started the rest of the review. Would you take a look? Thanks

Reply:
I think it might be a common mistake to enter "3" instead of 3, although it doesn't seem to matter since the "3" gets converted to a 3 anyway. 

I'll update it to use the js 'Infinity' (no quotes). 

But it seems like a validation error that a string is accepted and will be converted to mean 'Infinity' (unless it can be converted to int). Do you want me to open a ticket for it?

Reply:
https://jira.mongodb.org/browse/MONGOSH-758